From: Lukas Nykryn <lnykryn@redhat.com>
Date: Fri, 21 Nov 2014 17:02:15 +0100
Subject: selinux-access: fix broken ternary operator

it seems to be a typo introduced by ebcf1f97de4f6b1580ae55eb56b1a3939fe6b602
-                _r = selinux_access_check(_b, _m, _u->source_path ?:_u->fragment_path, (permission), &_error); \
+        ({ Unit *_unit = (unit); selinux_generic_access_check(bus,message, _unit->fragment_path ?: _unit->fragment_path, permission,error); })
---
 src/core/selinux-access.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/selinux-access.h b/src/core/selinux-access.h
index 27d9e14..fea8a85 100644
--- a/src/core/selinux-access.h
+++ b/src/core/selinux-access.h
@@ -37,7 +37,7 @@ int selinux_generic_access_check(sd_bus_message *message, const char *path, cons
 #define selinux_unit_access_check(unit, message, permission, error) \
         ({                                                              \
                 Unit *_unit = (unit);                                   \
-                selinux_generic_access_check((message), _unit->fragment_path ?: _unit->fragment_path, (permission), (error)); \
+                selinux_generic_access_check((message), _unit->source_path ?: _unit->fragment_path, (permission), (error)); \
         })
 
 #else
